test/functional/shell/viml_system_spec.lua

```diff
@@ -184,14 +184,16 @@ describe('system()', function()
     end)
   end)
 
-  if xclip then
     describe("with a program that doesn't close stdout", function()
-      it('will exit properly after passing input', function()
-        eq('', eval([[system('xclip -i -selection clipboard', 'clip-data')]]))
-        eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
-      end)
+      if xclip then
+        it('will exit properly after passing input', function()
+          eq('', eval([[system('xclip -i -selection clipboard', 'clip-data')]]))
+          eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
+        end)
+      else
+        pending('skipped (missing xclip)')
```

Comment:
Although I normally favor positive logic ("if true ... else"), for validation I think it's useful to have the "short block" near the top:

```lua
if not xclip then
  pending(...)
else
  -- test logic ...
end
```

What do you think?